src/Algorithm.ts

```diff
@@ -9,22 +11,61 @@ export default class Algorithm extends Builder {
     context.inAlg = true;
     const { node } = context;
 
-    if ('ecmarkdownOut' in node) {
-      // i.e., we already parsed this during the lint phase
-      // @ts-ignore
-      node.innerHTML = node.ecmarkdownOut.replace(/((?:\s+|>)[!?])\s+(\w+\s*\()/g, '$1&nbsp;$2');
-      return;
-    }
+    // prettier-ignore
+    const rawHtml =
+      'ecmarkdownOut' in node
+        ? (node as any).ecmarkdownOut
+        : emd.algorithm(node.innerHTML);
 
     // replace spaces after !/? with &nbsp; to prevent bad line breaking
-    const html = emd
-      .algorithm(node.innerHTML)
-      .replace(/((?:\s+|>)[!?])\s+(\w+\s*\()/g, '$1&nbsp;$2');
+    const html = rawHtml.replace(/((?:\s+|>)[!?])\s+(\w+\s*\()/g, '$1&nbsp;$2');
     node.innerHTML = html;
+
+    let labeledStepEntries: StepBiblioEntry[] = [];
+    let replaces = node.getAttribute('replaces-step');
+    if (replaces) {
+      replaces = 'step-' + replaces;
+      context.spec.replacementAlgorithms.push({
+        element: node,
+        target: replaces,
+      });
+      context.spec.replacementAlgorithmToContainedLabeledStepEntries.set(node, labeledStepEntries);
+    }
+
+    let labeledSteps = Array.from(node.querySelectorAll('li[id]'));
+    if (replaces && labeledSteps.length > 0 && node.firstElementChild!.children.length > 1) {
+      logWarning(
+        'You should not label a step in an replacement algorithm which has multiple top-level steps because the resulting step number could be ambiguous.'
```

Comment:
```suggestion
        'You should not label a step in a replacement algorithm which has multiple top-level steps because the resulting step number could be ambiguous.'
```